### PR TITLE
Include Change Log in Client Packages Nuget

### PR DESCRIPTION
**Pull request by chidozieononiwu**
Add CHANGELOG.md to Dataplane packages if they exist.
Include CHANGELOG.md in (Track 2) Client Library Package Directory.

-Link to change log will be included in nuget packages and show up on nuget.com
-Change Log will also show up as release notes in github releases as they do int he python repo. https://github.com/Azure/azure-sdk-for-python/releases
Change Log guidance can be found [here](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#changelog-guidance)

### CODE REVIEW

**Review by jsquire — APPROVED** [review]
The changes for Event Hubs track two look good to me.   I wonder, however, if we should be making these changes to the track one projects for those that the respective service teams continue to own and drive.

**Comment by pakrym** [comment]
Agree with @jsquire, if we are not positive that there would be content added to track 1 changelogs we should not include the empty files and scope this change to track 2 projects

**Review by heaths — APPROVED** [review]
Approved on KV renames, and thank you! The non-standard casing was scratching at my mild OCD.

**Comment by tg-msft** [comment]
Do we also need to make any changes to ensure these are properly formatted?  Storage also links to Changelog.txt from the package info in the various .csproj so those links should also be updated if we're changing the file names.

File: eng/Directory.Build.Data.targets [hunk 1]
```diff
@@ -10,15 +10,19 @@
...
    '$(IsSamplesProject)' != 'true' and '$(IsTestSupportProject)' != 'true' and '$(CommitSHA)' != ''">
     <PropertyGroup>
       <CurrentCsprojPath>$(MSBuildProjectDirectory)</CurrentCsprojPath>
       <DirectoryPartofPath>$(CurrentCsprojPath.Substring($(CurrentCsprojPath.LastIndexOf("\sdk\"))))</DirectoryPartofPath>
       <DirectoryPartofPath>$(DirectoryPartofPath.Remove($(DirectoryPartofPath.LastIndexOf("\src"))))</DirectoryPartofPath>
       <PackageProjectUrl>$(RepositoryUrl)/blob/$(CommitSHA)$(DirectoryPartofPath)/README.md</PackageProjectUrl>
+      <PackageReleaseNotes>$(RepositoryUrl)/blob/$(CommitSHA)$(DirectoryPartofPath)/CHANGELOG.md<PackageReleaseNotes>
     </PropertyGroup>
+    <ItemGroup>
+      <None Include="$(RepoRoot)/$(DirectoryPartofPath)/CHANGELOG.md" Pack="true" PackagePath="changelog.md"/>
```
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: We should probably add some exists checks here and only include the file and the link if the file exists.

**Comment by weshaggard** [comment]
Looks like you added one to `sdk/keyvault/CHANGELOG.md` which I don't think you wanted. They should be under the package directory.

**Comment by weshaggard** [comment]
>Do we also need to make any changes to ensure these are properly formatted? Storage also links to Changelog.txt from the package info in the various .csproj so those links should also be updated if we're changing the file names.

As part of the versioning work that @danieljurek and @chidozieononiwu  are working on we would like to also automatically add a correctly formatted header for any new versions we update in the projects.

**Comment by tg-msft** [comment]
>  automatically add a correctly formatted header for any new versions

Can we update the PR to include a little more context about what that'll look like, how we're supposed to edit these files so it doesn't get in the way, etc.?

**Comment by nemakam** [comment]
Looks good wrt ServiceBus track-1 side.
There is still development happening and we would like to have the Changelog integration for the track-1 sdk as well.

**Comment by weshaggard** [comment]
>Can we update the PR to include a little more context about what that'll look like, how we're supposed to edit these files so it doesn't get in the way, etc.?

@tg-msft we don't have that implemented yet so don't stress much about it right now but the goal will be to add a header with the new version and a blank section under whenever our tools update the package versions in the repo. However if what you are looking for is the format of the changelog have a look at https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#changelog-guidance.

File: eng/Directory.Build.Data.targets (hunk 1 above)
Comment (heaths) [reply eng/Directory.Build.Data.targets]: Also, the file should be named CHANGELOG.md in the package. The standard convention is all uppercase sans file extension, so should be stored in teh package as such.

**Comment by chidozieononiwu** [comment]
azp run net - client - ci

**Pull request merged**